contract/r/gnoswap/referral/keeper_test.gno

```diff
@@ -836,3 +840,57 @@ func TestTryRegisterRetryAfterFailure(t *testing.T) {
 		t.Error("should fail due to rate limit")
 	}
 }
+
+func TestRegisterWithEmptyRefAddrAndThenValidRefAddr(t *testing.T) {
+	cleanup := mockValidCaller()
+	defer cleanup()
+
+	// try to register with empty refAddr
+	success := TryRegister(validAddr5, "")
+	if !success {
+		t.Error("empty refAddr registration should succeed")
```

Comment:
error message :
I think it's better to have another error message like "empty refAddr registration should be bypassed"

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/620/commits/74425fcaba254e23a0e72a62c44f85a7cc3f317b